client-src/elements/chromedash-featurelist.ts

```diff
@@ -8,48 +10,47 @@ const MAX_FEATURES_SHOWN = 500;
   static styles = FEATURELIST_CSS;
 
-  static get properties() {
-    return {
-      user: {type: Object},
-      isSiteEditor: {type: Boolean},
-      signedInUser: {type: String},
-      editableFeatures: {type: Object},
-      features: {attribute: false}, // Directly edited and accessed in template/features.html
-      metadataEl: {attribute: false}, // The metadata component element. Directly edited in template/features.html
-      searchEl: {attribute: false}, // The search input element. Directly edited in template/features.html
-      filtered: {attribute: false},
-      openFeatures: {attribute: false},
-      starredFeatures: {attribute: false},
-    };
-  }
-
-  constructor() {
-    super();
-    this.user = {};
-    this.features = [];
-    this.editableFeatures = [];
-    this.filtered = [];
-    this.metadataEl = document.querySelector('chromedash-metadata');
+  @property({type: Object})
+  user!: User;
+  @property({type: Boolean})
+  isSiteEditor = false;
+  @property({type: String})
+  signedInUser = '';
+  @property({type: Array})
+  editableFeatures!: Feature[];
+  @property({type: Object})
+  metadataEl; // The metadata component element. Directly edited in template/features.html
+  @property({attribute: false})
+  _onFeatureToggledBound;
+  @property({attribute: false})
+  _onStarToggledBound;
+  @state()
+  features!: Feature[]; // Directly edited and accessed in template/features.html
+  @state()
+  searchEl; // The search input element. Directly edited in template/features.html
+  @state()
+  filtered!: Feature[];
+  @state()
+  openFeatures = new Set<number>();
+  @state()
+  starredFeatures = new Set<number>();
+  @state()
+  _hasInitialized = false; // Used to check initialization code.
+  @state()
+  _hasScrolledByUser = false; // Used to set the app header state.
+  @state()
+  _featuresUnveilMetric = new Metric('features_unveil');
+  @state()
+  _featuresFetchMetric = new Metric('features_loaded');
+
+  connectedCallback() {
+    super.connectedCallback();
     this.searchEl = document.querySelector('.search input');
-    this.isSiteEditor = false;
-    this.signedInUser = '';
-    this._hasInitialized = false; // Used to check initialization code.
-    this._hasScrolledByUser = false; // Used to set the app header state.
-    /* When scrollTo(), we also expand the feature. This is the id of the feature. */
-    this.openFeatures = new Set();
-    this.starredFeatures = new Set();
-
-    this._featuresUnveilMetric = new Metric('features_unveil');
-    this._featuresFetchMetric = new Metric('features_loaded');
-    this._featuresUnveilMetric.start();
+    this.metadataEl = document.querySelector('chromedash-metadata');
     this._featuresFetchMetric.start();
-
-    /* The running context `this` inside `renderItem` is `lit-virtualizer`
-     * rather than `chromedash-featurelist`, so all function calls inside need
-     * to be bound to `this`. */
+    this._featuresUnveilMetric.start();
```

Comment:
You should probably keep these `.start()` calls in the constructor, to avoid changing any timings in this PR. It's possible that `connectedCallback()` is actually a better place, but that bug fix should be separate from the Typescript conversion.